test/app-tap/datetime.test.lua

```diff
@@ -1867,6 +1868,51 @@ test:test("totable{}", function(test)
     end
 end)
 
+test:test('totable{} with timezone', function(test)
+    test:plan(6)
+
+    local DEFAULT_TZOFFSET = 0
+    local DEFAULT_TZ = ''
+
+    local MOSCOW_TZOFFSET = 180
+    local MOSCOW_TZ = 'Europe/Moscow'
+
+    local test_cases = {
+        -- Empty timezone values except tz.
+        {
+            dt = {tz = MOSCOW_TZ},
+            expected = {
+                tzoffset = MOSCOW_TZOFFSET,
+                tz = MOSCOW_TZ,
+            }
+        },
+        -- Empty timezone values except tzoffset.
+        {
+            dt = {tzoffset = MOSCOW_TZOFFSET},
+            expected = {
+                tzoffset = MOSCOW_TZOFFSET,
+                tz = '',
+            }
+        },
+        -- Empty timezone values.
+        {
+            dt = {},
+            expected = {
+                tz = DEFAULT_TZ,
+                tzoffset = DEFAULT_TZOFFSET,
+            },
+        },
+    }
+
+    for _, tc in pairs(test_cases) do
+        local dt_table = date.new(tc.dt):totable()
+        test:is(dt_table.tzoffset, tc.expected.tzoffset,
+                ('[tzoffset]: %s == %s'):format(dt_table.tzoffset, tc.expected.tzoffset))
```

Comment:
I suggest using `%q` here and below to avoid misleading messages in the case of an error:
```
not ok - [tz]: nil == 
```

Reply:
Fixed:

```diff
--- a/test/app-tap/datetime.test.lua
+++ b/test/app-tap/datetime.test.lua
@@ -1907,7 +1907,8 @@ test:test('totable{} with timezone', function(test)
     for _, tc in pairs(test_cases) do
         local dt_table = date.new(tc.dt):totable()
         test:is(dt_table.tzoffset, tc.expected.tzoffset,
-                ('[tzoffset]: %s == %s'):format(dt_table.tzoffset, tc.expected.tzoffset))
+                ('[tzoffset]: %s == %q'):format(dt_table.tzoffset,
+                                                tc.expected.tzoffset))
         test:is(dt_table.tz, tc.expected.tz,
                 ('[tz]: %s == %s'):format(dt_table.tz, tc.expected.tz))
     end
```

Reply:
I suppose it is better to use `%q` for the "got" argument too, since it can be `''`.

It looks like I made a mistake and referenced the wrong line (I mean the one with `[tz]`). But if the `tzoffset` may be an empty string too, it is better to use `%q` here too.

Reply:
Updated (`%q` for both test asserts):

```diff
     for _, tc in pairs(test_cases) do
         local dtab = date.new(tc.dt):totable()
         local expected = tc.expected
         test:is(dtab.tzoffset, expected.tzoffset,
-                ('[tzoffset]: %s == %q'):format(dtab.tzoffset,
+                ('[tzoffset]: %q == %q'):format(dtab.tzoffset,
                                                 expected.tzoffset))
         test:is(dtab.tz, expected.tz,
-                ('[tz]: %s == %s'):format(dtab.tz, expected.tz))
+                ('[tz]: %q == %q'):format(dtab.tz, expected.tz))
     end
 end)
 ```